File: eftl/sdk/java/src/com/tibco/eftl/websocket/WebSocketListener.java
/*
 * Copyright (c) 2013-$Date$ Cloud Software Group, Inc.
 * Licensed under a BSD-style license. Refer to [LICENSE]
 *
 * $Id$
 *
 */
package com.tibco.eftl.websocket;

public interface WebSocketListener {

    public void onOpen();
    
    public void onClose(int code, String reason);
    
    public void onError(Throwable cause);

    public void onMessage(String text);
    
    public void onMessage(byte[] data, int offset, int length);
    
    public void onPong(byte[] data, int offset, int length);
}

File: eftl/sdk/dotnet/TIBCO.EFTL/websocket/WebSocketListener.cs
/*
 * Copyright (c) 2013-$Date$ Cloud Software Group, Inc.
 * Licensed under a BSD-style license. Refer to [LICENSE]
 *
 * $Id$
 *
 */
using System;
using System.Collections;
using System.Linq;
using System.Text;

namespace TIBCO.EFTL
{
    public interface WebSocketListener 
    {
        void OnOpen();
        
        void OnClose(int code, String reason);
    
        void OnError(Exception cause);
        
        void OnMessage(String text);
        
        void OnMessage(byte[] data, int offset, int length);
        
        void OnPong(byte[] data, int offset, int length);
    }
}
